internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -316,27 +325,211 @@ func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m
 		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS cluster entry, name : %s", d.Get(NameKey)))
 	}
 
+	opsRetryTimeout := getRetryTimeout(d)
+
 	clusterSpec := constructSpec(d)
 
 	// TODO: update nodepools seperatly
+	err = handleNodepoolDiffs(config, opsRetryTimeout, getResp.EksCluster.FullName, clusterSpec.NodePools)
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control EKS cluster's nodepools, name : %s", d.Get(NameKey)))
+	}
+
 	clusterSpec.NodePools = nil
 
-	getResp.EksCluster.Meta = common.ConstructMeta(d)
-	getResp.EksCluster.Spec = clusterSpec
+	err = handleClusterDiff(config, getResp.EksCluster, common.ConstructMeta(d), clusterSpec)
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control EKS cluster entry, name : %s", d.Get(NameKey)))
+	}
+
+	log.Printf("[INFO] cluster update successful")
+
+	return dataSourceTMCEKSClusterRead(ctx, d, m)
+}
+
+func handleClusterDiff(config authctx.TanzuContext, tmcCluster *eksmodel.VmwareTanzuManageV1alpha1EksclusterEksCluster, meta *objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta, clusterSpec *eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec) error {
+	updateCluster := false
+
+	if meta.Description != tmcCluster.Meta.Description ||
+		!reflect.DeepEqual(meta.Labels, tmcCluster.Meta.Labels) {
+		updateCluster = true
+		tmcCluster.Meta.Description = meta.Description
+		tmcCluster.Meta.Labels = meta.Labels
+	}
+
+	if !reflect.DeepEqual(clusterSpec, tmcCluster.Spec) {
+		updateCluster = true
+	}
+
+	// The TF update request was only for nodepools.
+	// No need to update Cluster, it will error out.
+	if !updateCluster {
+		return nil
+	}
+
+	// there is some translation error, which results
+	// in mismatch on the server.
+	tmcCluster.Meta.CreationTime = strfmt.DateTime{}
```

Comment:
ideally in the update, the meta attributes like creation time, uid etc should not be updated. 
even if the client provides an updated value, the changes should not be honoured

Reply:
Yes, ideally we should. But the TMC API here checks if the update is set. If set, it checks that it is not modified. Without unsetting it, we get flaky errors when the conversion is not accurate and the time on the server doesn't match the time we pass.